hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -921,6 +930,26 @@ private void initializeFileGroups(HoodieTableMetaClient dataMetaClient, Metadata
     }, fileGroupFileIds.size());
   }
 
+  /**
+   * Returns a log creation hook impl.
+   */
+  private LogFileCreationCallback getLogCreationCallback(HoodieStorage storage,
+                                                         String instantTime,
+                                                         String partitionPath,
+                                                         String fileId) {
+    return new LogFileCreationCallback() {
+      @Override
+      public boolean preFileCreation(HoodieLogFile logFile) {
+        WriteMarkers writeMarkers = new DirectWriteMarkers(
```

Comment:
Maybe we should add a utility method to `WriteMarkersFactory.getDirectMarkers` ?

Reply:
I can be merged now.